ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java

```diff
@@ -0,0 +1,365 @@
+    for (Long usedCapacity : usedCapacities) {
+      blobStore.usedCapacity = usedCapacity;
+      if (blobStore.usedCapacity < (config.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0
+          * blobStore.capacityInBytes)) {
+        verifyCompactionDetails(null, blobStore, compactionPolicy);
+      } else {
+        verifyCompactionDetails(new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, bestCandidates),
+            blobStore, compactionPolicy);
+      }
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for
+   * {@link StoreConfig#storeMinUsedCapacityToTriggerCompactionInPercentage}
+   */
+  @Test
+  public void testDifferentThresholdsForMinLogSizeToCompact() throws StoreException, InterruptedException {
+
+    int[] minLogSizeToTriggerCompactionInPercentages = new int[]{10, 20, 35, 40, 50, 59, 60, 61, 65, 70, 80, 95};
+    // when used capacity(60%) is <= (minLogSize) % of total capacity, compactionDetails is expected to be null.
+    // If not, logSegmentsNotInJournal needs to be returned
+    List<String> bestCandidates = null;
+    for (int minLogSize : minLogSizeToTriggerCompactionInPercentages) {
+      initializeBlobStore(properties, time, minLogSize, -1, DEFAULT_MAX_BLOB_SIZE);
+      if (type.equals(CompactionPolicyType.DEFAULT)) {
+        bestCandidates = setUpStateForDefaultCompactionPolicy(blobStore, mockBlobStoreStats);
+      } else if (type.equals(CompactionPolicyType.COMPACT_ALL)) {
+        blobStore.logSegmentsNotInJournal = generateRandomStrings(3);
+        bestCandidates = blobStore.logSegmentsNotInJournal;
+      }
+      if (blobStore.usedCapacity < (config.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0
+          * blobStore.capacityInBytes)) {
+        verifyCompactionDetails(null, blobStore, compactionPolicy);
+      } else {
+        verifyCompactionDetails(new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, bestCandidates),
+            blobStore, compactionPolicy);
+      }
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for
+   * {@link StoreConfig#storeDeletedMessageRetentionDays}
+   */
+  @Test
+  public void testDifferentMessageRetentionDays() throws StoreException, InterruptedException {
+    List<String> bestCandidates = null;
+    int[] messageRetentionDayValues = new int[]{1, 2, 3, 6, 9};
+    for (int messageRetentionDays : messageRetentionDayValues) {
+      time = new MockTime();
+      Pair<MockBlobStore, StoreConfig> initState =
+          initializeBlobStore(properties, time, -1, messageRetentionDays, DEFAULT_MAX_BLOB_SIZE);
+      if (type.equals(CompactionPolicyType.DEFAULT)) {
+        bestCandidates = setUpStateForDefaultCompactionPolicy(blobStore, mockBlobStoreStats);
+        compactionPolicy = new DefaultCompactionPolicy(initState.getSecond(), time);
```

Comment:
we initialize a new StoreConfig within the for loop. and hence need to construct a new compactionPolicy